finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4ServerDispatcher.scala

```diff
@@ -123,14 +123,17 @@ class Netty4ServerDispatcher(
    * unexpected frame.  Unexpected frames cause the
    * connection to be closed with a protocol error.
    */
-  override protected[this] def demuxNewStream(f: Http2StreamFrame): Future[Unit] = f match {
+  override protected[this] def demuxNewStream(f: Http2Frame): Future[Unit] = f match {
     case frame: Http2HeadersFrame =>
-      val st = newStreamTransport(frame.streamId)
+      val st = newStreamTransport(frame.stream.id)
       if (st.recv(frame)) serveStream(st)
       Future.Unit
 
+    case settingsFrame: Http2SettingsFrame =>
```

Comment:
I believe this passes the settingsFrame forward to the remote so that it can deal with the remote can deal with the frame accordingly. When working through this change, the `Netty4ServerDispatcher` would receive this frame and then throw an exception because saw it as unexpected. I checked to see if the same behavior happened before the Netty upgrade and it seemed to work fine. I believe the new netty version reworked the class hierarchy so that the `settingsFrame was no longer a `Http2StreamFrame` and made it as a subclass of `Http2Frame`.